packages/frontend/src/components/Explorer/VisualizationCard/VisualizationCard.tsx

```diff
@@ -55,24 +54,23 @@ const VisualizationCard: FC<{
     const setFetchAll = useExplorerContext(
         (context) => context.queryResults.setFetchAll,
     );
-    const queryResults = useExplorerContext(
```

Comment:
Explorer context is constantly changing, causing this to generate a new object reference. 
Refactored this to only get the queryResults and then create the object if that changed.
Same fix done on `packages/frontend/src/pages/MinimalSavedExplorer.tsx`